bus-menu-scripts/mess.py

```diff
@@ -0,0 +1,210 @@
+import json
+import gspread
+import numpy
+import string
+import datetime
+#-------------------------------------------------------------------------------
+# What week is now? (1/2/3/4) (for deciding which menu to use)
+d = datetime.date.today()
+current_week = d.isocalendar()[1] - datetime.date(d.year,d.month,1).isocalendar()[1] + 1
+
+#-------------------------------------------------------------------------------
+# Get sheets
+
+gc = gspread.oauth()
+sh = gc.open_by_url("https://docs.google.com/spreadsheets/d/13R9rHEto70_CeFNH8e5Sybj5oYaz0WOBl_lY1an8P2U")
+#sh = gc.open_by_url("https://docs.google.com/spreadsheets/d/1ZiprHJUM98G6yS-aN1h5Vt1IJLjDavuDmJL-dxVaEsc")
+weekly = sh.worksheet("Aug'22")
+additionals = sh.worksheet("Extras")
+
+#-------------------------------------------------------------------------------
+# finding correct worksheet
+while True:
+    try:
+        weekly = sh.worksheet(d.strftime("%b'%y"))
+    except:
+        if (d.month>=11):
+            d = d.fromisoformat(d.strftime("%Y-{}-%d".format(d.month-1)))
+        else:
+            d = d.fromisoformat(d.strftime("%Y-0{}-%d".format(d.month-1)))
+    else:
+        break
+
+weekly_grid = numpy.array(weekly.get_all_values())
+additionals_grid = numpy.array(additionals.get_all_values())
+
+#-------------------------------------------------------------------------------
+# number of coloumns for each meal
+breakfast = weekly.find("Breakfast").col
+lunch = weekly.find("Lunch").col
+snacks = weekly.find("Snacks").col
+dinner = weekly.find("Dinner").col
+
+space={"Breakfast":lunch-breakfast,"Lunch":snacks-lunch,"Snacks":dinner-snacks,"Dinner":weekly_grid.shape[1]-snacks}
+
+total_space = sum(space.values())
+
+#-------------------------------------------------------------------------------
+# Build object structure
+
+days = ["Sunday", "Monday", "Tuesday",
+        "Wednesday", "Thursday", "Friday", "Saturday"]
+meals = ["Breakfast", "Lunch", "Snacks", "Dinner"]
+
+regular_items = {}
+extra_items = {}
+daily_items = {}
+
+for day in days:
```

Comment:
You can do this in a slightly better fashion using `dict.fromkeys()`

Something like `regular_items = dict.fromkeys(days,[])` might work

Reply:
Previously I had put `reg_items = dict.fromkeys(days,[])`, but the problem is that this creates a shallow copy of `[]` resulting in addition of an item in every value of the dictionary

Reply:
Oh yeah, nice catch.

You can use dict comprehension instead. Something like `regular_items = {day:[] for day in days}`

Reply:
Added the dict comprehensions.